docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+# Components
+
+This section explains the different components of KPOps, their usage and
+configuration. To learn more about their hierarchy, visit the
+[architecture](/docs/developer/architecture/component-inheritance.md) page.
+
+## KubernetesApp
+
+### Usage
+
+Can be used to deploy any app in Kubernetes using Helm, for example, a REST
+service that serves Kafka data.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kubernetes-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Deploy using Helm.
+
+#### destroy
+
+Uninstall Helm release.
+
+#### reset
+
+Not implemented.
+
+#### clean
+
+Not implemented.
+
+## KafkaApp
+
+Sub class of [_KubernetesApp_](#kubernetesapp).
+
+### Usage
+
+- Deploys a [streams-bootstrap](https://github.com/bakdata/streams-bootstrap)
+component
+- Should not be used in `pipeline.yaml` as the component can be defined as either a
+[StreamsApp](#streamsapp) or a [Producer](#producer)
+- Often used in `defaults.yaml`
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kafka-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+In addition to [KubernetesApp's](#kubernetesapp) `deploy`:
+
+- Create topics if provided (optional)
+- Submit schemas to the registry if provided (optional)
```

Comment:
```suggestion
- Submit Avro schemas to the registry if provided (optional)
```

Reply:
Or we should somewhere mention this limitation of not supporting ProtoBuf

Reply:
Hmm, I have a feeling that it belongs to another (non-existent as of yet) page, but for now adding it here is good I think. I will also include it into the Configuration example.

Reply:
Yeah, I agree with you on that. I was just thinking loudly. Maybe we should have a limitation section.

Reply:
If there are enough limitations that we can come up with, sure